src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -93,27 +93,12 @@ def __init__(
         self.sequence_length = sequence_length
         self.sampling_temperature = sampling_temperature
         self.deterministic = deterministic
+        self.use_deepsparse_cache = use_deepsparse_cache
         self.input_ids_length = input_ids_length
         self.kv_cache_enabled = kv_cache_enabled
-        self.kv_cache = (
-            DecoderKVCache(use_deepsparse_cache) if kv_cache_enabled else None
-        )
+        self.kv_cache_storage = SessionStorageKVCache() if kv_cache_enabled else None
+        self.session = None
```

Comment:
session cannot be a property of the pipeline - it must exist only in the context of an inference after being grabbed from storage. if we're reading and writing sessions from the object then we run into race conditions when multiple inferences are run at the same time

Reply:
Please read carefully, this is not the pipeline attribute, this is the attribute of a `NLDecoderEngine`.